### PR TITLE
adding UI Tester test/example for ButtonEditor_simple_demo.py

### PR DESCRIPTION
**Pull request by aaronayres35**
This is the first "example" (which is actually a test of a traitsui example/demo) to be added for UI Tester. 
Addresses part of #1220

### CODE REVIEW

File: integrationtests/test_all_examples.py [hunk 1]
```diff
@@ -346,11 +346,26 @@ def test_run(self):
...
+        )
+        demo = load_demo(filepath, "demo")
+
+        tester = UITester()
+        with tester.create_ui(demo) as ui:
+            button = tester.find_by_name(ui, "my_button_trait")
+            for _ in range(5):
+                button.perform(command.MouseClick())
+            click_counter = tester.find_by_name(ui, "click_counter")
+            displayed_count = click_counter.inspect(query.DisplayedText())
```
Comment (aaronayres35) [inline integrationtests/test_all_examples.py:366]: There is still the issue on OSX/qt where the click_counter does not actually display a change, however this test still passes... Seems a little misleading given the test, but I think the tester, and query.DisplayedText() are generally working how we want them to.

File: integrationtests/test_all_examples.py [hunk 2]
```diff
@@ -346,11 +346,26 @@ def test_run(self):
...
+        filepath = os.path.join(
+            DEMO, "Standard_Editors", "ButtonEditor_simple_demo.py"
+        )
+        demo = load_demo(filepath, "demo")
+
+        tester = UITester()
+        with tester.create_ui(demo) as ui:
+            button = tester.find_by_name(ui, "my_button_trait")
+            for _ in range(5):
+                button.perform(command.MouseClick())
```
Comment (kitchoi) [inline integrationtests/test_all_examples.py:364]: I'd suggest adding `self.assertEqual(demo.click_counter, index + 1)` following the mouse click.
This demonstrates one of the two common testing patterns: Modify GUI state resulting in trait changes

Likewise, adding `demo.click_counter = 10` (deliberately not 5) and then inspect the displayed value is updated to "10" demonstrates the second of the two common testing patterns: Modify trait updates the GUI state.

**Review by kitchoi — COMMENTED** [review]
LGTM. Some suggestions.

**Review by kitchoi — APPROVED** [review]
LGTM

**Pull request merged**